components/net/tests/main.rs

```diff
@@ -95,6 +95,25 @@ fn create_embedder_proxy() -> EmbedderProxy {
     }
 }
 
+fn create_http_state() -> HttpState {
```

Comment:
I was able to remove the default implementation for `HttpState` and created this helper function that does the same thing but is only available in tests. Fortunately there was a `create_embedder_proxy` function which creates a dummy `embedder_proxy`. Thanks to this, there is no need for making `embedder_proxy` optional in `HttpState`